src/main/java/com/google/cloud/genomics/utils/ShardUtils.java

```diff
@@ -232,29 +233,37 @@ public SearchReadsRequest apply(Contig shard) {
   }
 
   /**
-   * Constructs StreamReadsRequest for the readGroupSetIds, assuming that the user wants to
-   * include all references.
+   * Constructs sharded StreamReadsRequest for all references in the referenceSet associated
+   * with the readGroupSetId.
    * 
-   * TODO: Should this be sharded - by the referenceBounds for the associated referenceSet
-   *     and/or by read groups?
-   * 
-   * @param readGroupSetIds The readGroupSetIds.
+   * @param readGroupSetId The readGroupSetId.
+   * @param sexChromosomeFilter An enum value indicating how sex chromosomes should be
+   *        handled in the result.
+   * @param numberOfBasesPerShard The maximum number of bases to include per shard.
+   * @param auth The OfflineAuth to be used to get the reference bounds for the variantSet.
    * @return The shuffled list of sharded request objects.
+   * @throws IOException 
+   * @throws GeneralSecurityException 
    */
-  public static ImmutableList<StreamReadsRequest> getReadRequests(List<String> readGroupSetIds) {
-    // Work around lack of FluentIterable.shuffle() https://github.com/google/guava/issues/1358
-    List<StreamReadsRequest> requests =
-        Arrays.asList(FluentIterable.from(readGroupSetIds)
-        .transform(new Function<String, StreamReadsRequest>() {
+  public static ImmutableList<StreamReadsRequest> getReadRequests(final String readGroupSetId,
+      SexChromosomeFilter sexChromosomeFilter, long numberOfBasesPerShard,
```

Comment:
Feels like there may be more use cases for custom Filters. No need to do now, but if you think other filters make sense here, consider changing to a Filter superclass and dealing with a list of filters that the user can apply.